pkg/catalog/catalog_test.go

```diff
@@ -10,6 +10,46 @@ import (
 	"google.golang.org/protobuf/types/known/timestamppb"
 )
 
+func TestGetStartPos(t *testing.T) {
+	/**
+	Here are the possible states for where to start iterating, given a prefix, an "after" and a delimiter:
+	delim	prefix	after	expected start position
+	0		0		0		""
+	0		0		1		at after (when iterating we'll skip the exact match anyway)
+	0		1		0		at prefix
+	0		1		1		at after (when iterating we'll skip the exact match anyway)
+	1		0		0		""
+	1		0		1		next common prefix after "after"
+	1		1		0		at prefix
+	1		1		1		next common prefix after "after"
+	*/
+	cases := []struct {
+		Name      string
+		Delimiter string
+		Prefix    string
+		After     string
+		Expected  string
+	}{
+		{"all_empty", "", "", "", ""},
+		{"only_after", "", "", "a/b/", "a/b/"},
+		{"only_prefix", "", "a/", "", "a/"},
+		{"prefix_and_after", "", "a/", "a/b/", "a/b/"},
+		{"only_delimiter", "/", "", "", ""},
+		{"delimiter_and_after", "/", "", "a/b/", string(graveler.UpperBoundForPrefix([]byte("a/b/")))},
+		{"delimiter_and_prefix", "/", "a/", "", "a/"},
+		{"delimiter_prefix_and_after", "/", "a/", "a/b/", string(graveler.UpperBoundForPrefix([]byte("a/b/")))},
+	}
+
+	for _, cas := range cases {
+		t.Run(cas.Name, func(t *testing.T) {
+			got := GetStartPos(cas.Prefix, cas.After, cas.Delimiter)
+			if got != cas.Expected {
+				t.Fatalf("expected %s got %s", cas.Expected, got)
+			}
+		})
+	}
+}
+
 func TestCatalog_ListRepositories(t *testing.T) {
```

Comment:
Can/should we also add tests for listings with _empty_ prefixes, i.e. paths such as `a/b///c`, `a/b//c`, `a/b/c`?

Reply:
Yep, added